clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3619,4 +3619,14 @@ protected boolean isTransientRecordBufferUsed() {
   protected void setPartitionConsumptionState(int partition, PartitionConsumptionState pcs) {
     partitionConsumptionStateMap.put(partition, pcs);
   }
+
+  public void maybeBlockOnDrainerCapacity() {
+    while (storeBufferService.getTotalRemainingMemory() < 1000) {
```

Comment:
Should this 1000 be configurable? Or even if left un-configurable, should it at least be made into a constant?

Reply:
I was torn on adding another tunable parameter.  But maybe theres a happy middle?  Maybe instead of using a strict measure, we can change it to wait if it's like 90-95% at capacity?  That way it's at least tunable relative to the size of the configured drainer, but doesn't add extra noise.

Reply:
Why block at some margin below the actual limit? Can we instead block `if (remaining_capacity - incoming_payload_size < 0)` or something like that? This is not fully precise since the memory overhead of the transient record is more than just the payload size, but probably close enough to not matter?

Reply:
If the issue is due to drainer records being uncompressed and producer buffers are not. just checking drainer capacity still could lead very memory usage. We probably should make tunable and add a metric to check the transient cache usage.